cylc/flow/etc/job.sh

```diff
@@ -127,7 +127,7 @@ cylc__job__main() {
     export CYLC_TASK_WORK_PATH="${CYLC_TASK_WORK_DIR}"
 
     # Send task started message
-    cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" 'started' &
+    cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" 'DEBUG:started' &
```

Comment:
This addresses submitted/succeeded but not failed or custom messages.

(note there is additional logging for tasks which complete with unsatisfied required outputs so we don't need to go to extra effort to draw attention to the failed state any more)

Could we change the default for all `cylc message` calls rather than hardcoding just these two?

Reply:
You mean change the default in `cylc/flow/scripts/message.py`?

Reply:
Or wherever it's set, that way any user defined `cylc message` commands will also go to DEBUG level (unless they configure them otherwise).

Reply:
Keen to get @hjoliver 's thoughts on this

Reply:
Not sure about that.

It definitely makes sense for the standard job status messages, because they should (pretty much) always be followed by a logged status change.

But part of the intention for "custom messages" was to allow arbitrary non-triggering info to be sent to the scheduler log by tasks. (Whether or not anyone uses it that way, I'm not sure).  Debug would effectively turn that off for normal use.

And I think it's OK to log task failures even when they're handled by the graph. Failure is still, by definition, the task not doing what it is primarily supposed to do.

Reply:
Right, wasn't aware custom task messages were considered for non-triggering purposes.

Given the verbosity of workflow logs I highly doubt anyone is really using them in this way. Worth noting that custom outputs are now visible in the GUI. Also know at least one site is using custom messages to communicate model time-steps which would bloat the log somewhat. But can leave at INFO, I guess these messages wouldn't get logged otherwise so fair enough?

For failed tasks we would still get the task status changed message (`=> failed`) message right? So I don't think we need an additional log item telling you that the failed status was sent by `cylc message` (as opposed to being polled)?

Reply:
> Right, wasn't aware custom task messages were considered for non-triggering purposes.

Well if a message is received, we check to see if it corresponds to a task output or not. If not, log it anyway.

> For failed tasks we would still get the task status changed  ...

Agreed.

>  But can leave at INFO, I guess these messages wouldn't get logged otherwise so fair enough?

:+1: 

> Given the verbosity of workflow logs I highly doubt anyone is really using them in this way.

Yeah probably not, but there might still be potential use cases.

Reply:
> This addresses submitted/succeeded but not failed or custom messages.
>> Not sure about that.
>>> For failed tasks we would still get the task status changed ...
>>>> Agreed.

Are we agreed to remove this duplicate "failed" log entry then?

IMO each state transition should be logged once and only once as duplicate logging is making debugging quite difficult ATM.

Looks like the state transition message we are aiming to preserve are the `=> <state>` ones. So all others (excluding custom outputs) should go or be demoted to DEBUG if there's a good case to keep them.

Reply:
Yes agreed!

Reply:
It seems we can't demote the "failed" `cylc message` in the job script because then the `critical handlers` don't run (also we don't get `CRITICAL - failed/ERR` printed in `job.err`)

Reply:
Good point, that presumably also means the succeeded message would be missing too? Will have to handle server side rather than setting the log level explicitly in the job.sh.

Reply:
For
```
[runtime]
    [[foo]]
        script = echo Data
```
and `cylc play` in normal verbosity, I get this change in `job.out`:
```diff
 Data
-INFO - started
+DEBUG - started
-INFO - succeeded
+DEBUG - succeeded
```
So the succeeded message is still getting through.

> Will have to handle server side rather than setting the log level explicitly in the job.sh.

I think that would be fiddly, probably not going to attempt that in this PR

Reply:
I think it's probably just adding an `if message in {'started', 'running', 'succeeded', 'failed'}: severity = logging.DEBUG` before this block:

https://github.com/cylc/cylc-flow/blob/f88f32e2bf14568664d27f9452c655baa45eb9b7/cylc/flow/task_events_mgr.py#L811-L814